### ISSUE TITLE
promises being rejected with non-errors

### ISSUE THREAD

**Issue by yocontra** [phulst/node-tile38#10]
```
(node:98273) Warning: a promise was rejected with a non-error: [object String]
Type member is invalid. Expecting a string
```

just needs to be wrapped in an error class to get rid of these annoying warnings. would also be great if these errors had different classes, so you can differentiate between validation errors and connection errors.

**Comment by phulst** [comment]
I suppose rejecting promises with string values has been a bad habit of mine, but after reading some I agree that passing back different Error types would make it easier to differentiate and handle specific errors. 
I'll plan on updating this in a next release.. thx

**Comment by phulst** [comment]
I've committed a fix for this and now always rejecting with Error. There are only a few scenarios where a rejection may occur: 
* when the underlying Redis library returns an error. I'm just rejecting with whatever error I get back here
* when Tile38 returns an error response with ok=false and an 'err' property set. In this case I'm just rejecting with an Error, using the received err message. 
* If Tile38 returns malformed JSON. Also rejecting with Error here, and returning an error message plus the raw Tile38 response. 

In most of these cases (certainly in the first 2) it's not really feasible to figure out all possible errors that might be returned from Tile38 and/or Redis library and wrap these in custom Error objects, so keeping it simple and wrapping them in Error.

**Comment by phulst** [comment]
https://github.com/phulst/node-tile38/commit/9565e7cbaf659fe0f32b47b0ccfc99f394e8e855

**Comment by yocontra** [comment]
@phulst Thanks! Is this published on NPM?

**Comment by phulst** [comment]
yes, I just released 0.8.0 today
